### PR TITLE
8889 file-level PIDs configuration in individual collections

### LINKED ISSUE #8889: Feature Request: Enable file PIDs at Dataverse Collection Level
The Harvard Dataverse Model for supporting file-level DOIs will support collection level instead of production-wide file-level PIDs.
This requires the ability to enable file-level PIDs at the collection level. Eventually, we may discuss enabling at dataset level, if needed.
-->

**Overview of the Feature Request**

**What kind of user is the feature intended for?**
SUPERUSER/INSTALLATION ADMIN

**What inspired the request?**
THE COST OF PROVIDING FILE LEVEL PIDs FOR GENERALIST, OPEN REPOSITORIES

**What existing behavior do you want changed?**
MANAGED FILE LEVEL PIDs distribution

**Any brand new behavior do you want to add to Dataverse?**

**Any related open or closed issues to this feature request?**
NO

@scolapasta

### PR DESCRIPTION
**Pull request by landreev**
**What this PR does / why we need it**:
See issue. Curation team feature request, allows enabling file PIDs in specific collections. Enabling/disabling is via API only.  

**Which issue(s) this PR closes**:

- Closes #8889

**Special notes for your reviewer**:

The implementation is straightforward. I added a dedicated boolean flag to the Dataverse entity for this. I don't have any reservations about extending that table, especially since there should be relatively few entries in it. 

Note that I copy-and-pasted/stole the new "collection attributes" API from Oliver's #9462, he's adding it in order to be able to configure version PIDs in collections, makes perfect sense to use it for this as well. 

I added a simple integration test for the behavior. 

One extra thing: there is a Flyway script in the PR, I numbered it 5.13.0.2, under the assumption that it is going to make it into 5.14... but it may need to be renamed as needed. 

**Suggestions on how to test this**:

Straightforward. Have file PIDs disabled instance-wide. Enable them for a specific collection via the API:  `/api/dataverses/${ALIAS}/attribute/filePIDsEnabled?value=true`. Create a dataset with files in the collection, publish, confirm that the files have got PIDs. 
(Or the other way around - have it enabled for the instance, then disable it for a specific collection).
Important: needs to be tested on an instance using a real Datacite account. Because "FAKE" does not create file-level DOIs. 

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

Yes, included. 

**Additional documentation**:

### CODE REVIEW

File: doc/sphinx-guides/source/api/native-api.rst [hunk 1]
```diff
@@ -736,6 +736,24 @@ The fully expanded example above (without environment variables) looks like this
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: 
+
+  curl -H  X-Dataverse-key:$API_TOKEN $SERVER_URL/api/dataverses/$ID/attribute/$ATTRIBUTE?value=$VALUE
+
+The following attributes are supported:
+
+* ``alias``  Collection alias
+* ``name`` Name
```
Comment (pdurbin) [inline doc/sphinx-guides/source/api/native-api.rst:751]: This should be useful for @Kris-LIBIS.

In the containerization meetings he has expressed the need to change the name of a collection after installation via API.

File: src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java [hunk 2]
```diff
@@ -2091,6 +2091,12 @@ static Response setDataverseLogo(String dataverseAlias, String pathToImageFile,
                 .multiPart("file", new File(pathToImageFile))
                 .put("/api/dataverses/" + dataverseAlias + "/logo");
     }
+    
+    static Response setCollectionAttribute(String dataverseAlias, String attribute, String value, String apiToken) {
```
Comment (pdurbin) [inline src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:2095]: I'm excited to have this method and functionality. In future test cases, we could use it in DataversesIT to test changing the name, description, and affiliation of collections.

We COULD do that now, in this PR, if we feel like it. It's new functionality that should probably be exercised.

File: doc/sphinx-guides/source/api/native-api.rst [hunk 3]
```diff
@@ -736,6 +736,24 @@ The fully expanded example above (without environment variables) looks like this
...
+* ``description`` Description
+* ``affiliation`` Affiliation
+* ``filePIDsEnabled`` ("true" or "false") Enables or disables registraion of file-level PIDs in datasets within the collection (overriding the instance-wide setting).
```
Comment (pdurbin) [inline doc/sphinx-guides/source/api/native-api.rst:754]:
```suggestion
* ``filePIDsEnabled`` ("true" or "false") Enables or disables registration of file-level PIDs in datasets within the collection (overriding the instance-wide setting).
```

File: doc/sphinx-guides/source/installation/config.rst [hunk 4]
```diff
@@ -2756,14 +2756,16 @@ timestamps.
...
 ++++++++++++++++
 
-Toggles publishing of file-based PIDs for the entire installation. By default this setting is absent and Dataverse Software assumes it to be true. If enabled, the registration will be performed asynchronously (in the background) during publishing of a dataset.
+Toggles publishing of file-level PIDs for the entire installation. By default this setting is absent and Dataverse Software assumes it to be true. If enabled, the registration will be performed asynchronously (in the background) during publishing of a dataset.
 
 If you don't want to register file-based PIDs for your installation, set:
 
 ``curl -X PUT -d 'false' http://localhost:8080/api/admin/settings/:FilePIDsEnabled``
 
 Note: File-level PID registration was added in Dataverse Software 4.9; it could not be disabled until Dataverse Software 4.9.3.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/installation/config.rst:2765]:
```suggestion
```
Maybe it's time to remove this old note.

**Review by pdurbin — COMMENTED** [review]
I didn't test this but the code looks good and excited about the main functionality AND the new ability to modify collections via API for alias, name, description, and affiliation. A nice bonus feature.

I suggested some tiny stuff (typo fixes, maybe more tests) so I'll assign this to @landreev to take a look.

DownloadFilesIT.downloadAllFilesRestricted was failing when I got to this PR: https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-9614/5/testReport/edu.harvard.iq.dataverse.api/DownloadFilesIT/downloadAllFilesRestricted/

I must have bumped something accidentally triggering a new test run. 😅 I guess we'll see how it ends: https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-9614/6/

We should be mindful of the SQL script name, of course. It seems likely we can get this into 5.13 but we'll see.

File: src/main/java/edu/harvard/iq/dataverse/Dataverse.java [hunk 5]
```diff
@@ -590,8 +590,17 @@ public void setCitationDatasetFieldTypes(List<DatasetFieldType> citationDatasetF
         this.citationDatasetFieldTypes = citationDatasetFieldTypes;
     }
     
-    
+    @Column(nullable = true)
+    private Boolean filePIDsEnabled;
 
+    public Boolean getFilePIDsEnabled() {
+        return filePIDsEnabled;
```
Comment (poikilotherm) [inline src/main/java/edu/harvard/iq/dataverse/Dataverse.java:597]: Shouldn't this code return a default if null? If this should be optional (so this is handled as business logic in service layer), maybe return `Optional.ofNullable()`?

Comment (landreev) [reply src/main/java/edu/harvard/iq/dataverse/Dataverse.java:597]: I specifically wanted it to be nullable, and to be null by default; and I wanted the method above to return the actual value (or lack thereof) in the database table. Null value in my implementation has a meaning, of "not specifically configured" one way or another, indicating that the setting of the ancestral collection needs to be consulted instead (or the system-wide default, if none defined on the root collection). 
So yes, that was a long way of saying "business logic is elsewhere". I will add a javadoc entry.

File: src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java (hunk 2 above)
Comment (landreev) [reply src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:2095]: As I mentioned in the description, I'm copy-and-pasting/stealing this API from Oliver's pr that's still in draft (#9462). This by itself is going to introduce at least a minor merge conflict with his branch. @poikilotherm are you ok with me expanding on it in this branch, like adding an integration test? - or do you want me to leave it to you?

Comment (landreev) [reply src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:2095]: (I added a simple test that uses the API to change the alias then makes sure the change has taken effect)

Comment (poikilotherm) [reply src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:2095]: Please go ahead! Im happy to resolver any merge conflicts that might appear.

**Comment by landreev** [comment]
Is this PR ready now, or are you still waiting for me to add anything else? - I may have missed something.

**Review by pdurbin — APPROVED** [review]
Looks good!

**Comment by kcondon** [comment]
Issues found:
1. Getting 405 when calling the api endpoint: API endpoint does not support this method.
2. Need to confirm the current flyway script name is appropriate for this 5.14 release, per note in pr
3. Question: what is the expectation if system setting and collection setting are the same, true/true, false/false, seems obvious but just asking.
4. Question: how does an admin register file pids for datasets created before file pids was enabled? orig issues mentioned an api endpoint.

**Comment by landreev** [comment]
There's now an API for minting PIDs for files within a specific collection. `GET /api/admin/registerDataFiles/{collection_alias}`, described in the "Managing Datasets and Dataverse Collections" guide, with the other file-level registration methods.

**Pull request merged**